src/main/scala/org/ergoplatform/modifiers/history/ADProofs.scala

```diff
@@ -25,12 +26,10 @@ case class ADProofs(headerId: ModifierId,
 
   override val modifierTypeId: ModifierTypeId = ADProofs.modifierTypeId
 
-  override type M = ADProofs
-
-  override lazy val serializer: Serializer[ADProofs] = ADProofSerializer
-
   override def toString: String = s"ADProofs(Id:$id,HeaderId:$headerId)"
 
+  lazy val size: Int = sizeOpt.getOrElse(ADProofSerializer.toBytes(this).size)
```

Comment:
Let's return `BytesSerializable` interface back and move size back to parent class